lib/src/json_schema/validator.dart

```diff
@@ -512,7 +542,7 @@ class Validator {
   void _schemaDependenciesValidation(JsonSchema schema, Instance instance) {
     schema.schemaDependencies.forEach((k, otherSchema) {
       if (instance.data.containsKey(k)) {
-        if (!Validator(otherSchema).validate(instance)) {
+        if (Validator(otherSchema).validateWithResults(instance).errors.isNotEmpty) {
```

Comment:
```suggestion
        if (!Validator(otherSchema).validateWithResults(instance).isValid) {
```

Reply:
Damn I thought I had gotten all of these! Fail.